doc/sphinx/crystal_plasticity.rst

```diff
@@ -0,0 +1,293 @@
+
+.. note:: Future versions of NEML will replace the raw pointer math in the macroscale plasticity models with the object system.
+
+Storing history variables
+-------------------------
+
+The macroscale NEML material models use raw pointer arrays to represent the
+model internal variables.  This means the person implementing the model
+is responsible for remembering which variable aligns with which position in
+the flat array.  Similarly, it means that the end-user must know the
+index of an internal variable of interest in order to include it in the
+output of a finite element simulation.
+
+The history object stores the variables and corresponding meta-information --
+the name of the variable and the type of object (scalar, vector, rotation, etc.).
+The history object is described in a :ref:`seperate section <history>`.
+
+As with the tensor object system, the history set object can either manage its own memory or work with externally-managed memory.  Again, this facilitate vectorization and avoids copying when the input program can provide the history of many material points in a large block array.
+
+.. note:: Future versions of NEML will replace raw pointer arrays storing history vectors in the macroscale plasticity models with the object system.
+
+Polycrystal homogenization and plotting
+---------------------------------------
+
+To help with developing and debugging models, NEML provides python implementations of simple polycrystal homogenization models and a suite of python functions for plotting pole figures, inverse pole figures, and other crystallographic information.
+
+.. toctree::
+   cp/polycrystal.rst
+   cp/polefigures.rst
+
+Implementation
+--------------
+
+NEML crystal models are implemented as a subclass of the :ref:`large deformation incremental <NEMLModel_ldi>` interface.  In addition to the basic stress, history, and orientation updates the class also provides methods for getting the current crystal orientation (in active or passive convention), useful for output, and similar methods for setting or resetting the crystal orientation, useful either for bulk input of crystal orientations from a separate file for CPFEM calculations or for interfacing with models which cause crystal to reorientate, for example twinning or recrystallization models.
+
+The implementation uses a coupled, Euler implicit integration of the stress and internal variable rate equations.  
+After the model successfully updates these quantities it then uses a separate Euler explicit exponential integration of the elastic spin to update the crystal orientation.  The exponential integrator ensures the orientation remains in the special orthogonal group.
```

Comment:
General (dumb) question: Why do we need to integrate the vorticity to get the orientation, isn't R_e (the rotation of F_e) give the current orientation ? 
We don't generally need any orientation rate for constitutive laws, or, ?
Also, the continuum potato figure shows the different configuration, a thing which is not clear is does "our reference configuration" have the same sample and crystal coordinate (by multiplying the rotation based on current orientation) ? or do we keep track of them separately.

Reply:
We don't know F_e, just F or, the way I set this up, L.  You need to integrate the skew part of L, accounting for plastic spin, to get the elastic rotation, just like you need to integrate the symmetric part of L, accounting for plastic stretch, to get the elastic stretch (aka the Cauchy stress).

R_e does not give the full new orientation, instead it's the composition of R_e and the original crystal orientation that gives the full rotation from crystal to current.

The way that I have it setup we only keep track of the current crystal orientation Q.  You could just as well set it up to track R_e and Q_0 (the original grain orientation).